### ISSUE TITLE
[BUG] Failed connection Websocket

### ISSUE THREAD

**Issue by HoroshoVse** [androidseb25/iGotify-Notification-Assistent#98]
**Describe the bug**
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://[::]:8080
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: /app
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.
Disconnection happened, type: Error
Webseocket Reconnection failed with Error. Try to reconnect in 10s.

**Expected behavior**
Error when starting a container

**Screenshots**

**Smartphone (please complete the following information):**
 - Version of Container [e.g. 1.0.0.7]
 - Version of iGotify [e.g. 1.0.5]

**Additional context**

**Comment by androidseb25** [comment]
How did configure the container (docker compose with local environment or over the app)

Is the gotify container in you're network available and from the igotify container?

**Comment by doopler** [comment]
+ 1 for this issue.
here is my config:
version: '3.8'

services:
  gotify:
    container_name: gotify
    hostname: gotify
    image: gotify/server
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    networks:
      - net
    ports:
      - 8090:80
    volumes:
      - some_dir:/app/data
    environment:
      GOTIFY_DEFAULTUSER_PASS: some_password
      TZ: IST
      GOTIFY_UPLOADEDIMAGESDIR: data/images
      GOTIFY_PLUGINSDIR=: data/plugins

  igotify:
    container_name: igotify
    hostname: igotify
    image: ghcr.io/androidseb25/igotify-notification-assist:latest
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    pull_policy: always
    networks:
      - net
    ports:
      - 8091:8080
    volumes:
      -some_dir/igotify/app/data
    environment:
       GOTIFY_URLS: http://locahost:8090

networks:
  net:

volumes:
  data:
  api-data:

**Comment by sskokorin** [comment]
I have a similar problem.
I set everything up according to the guide and iGotify seems to work as expected, but I get these messages in the logs:
```
gotify   | 2024-10-12T16:50:40+03:00 | 401 |     238.304µs |   192.168.128.1 | GET      "/stream?token=[masked]"
gotify   | Error #01: you need to provide a valid access token or user credentials to access this api
igotify  | Disconnection happened, type: Error
igotify  | Webseocket Reconnection failed with Error. Try to reconnect in 10s.
```

**Comment by androidseb25** [comment]
Is this error still actually?

When no this Issue will be closed on day 30.

**Comment by Guezli** [comment]
I also have the problem that I get this error message in the iGotify container. 

I am using the Docker Compose template from Github. I have changed the port mapping and the bind of the folders.

`Disconnection happened, type: Error`
`Webseocket Reconnection failed with Error. Try to reconnect in 10s.`
`Gotify Server is not available try to reconnect in 10s.`
`Reconnecting...`

Even if I don't apply any changes to the template, the error persists. 

Docker is installed on an Intel Nuc on a Ubuntu server. The traffic goes through an NPM.

**Comment by androidseb25** [comment]
did you have enabled Websocket support on NPM?

**Comment by paractmol** [comment]
I also get this error. Websockets support is enabled

**Comment by Jan-Kruis** [comment]
Hi Sebastion,

I don’t quite understand your question about whether I want to configure a local or live instance.

mine gotify.domain.org is publicly accessible via the internet. My Traefik setup acts as a reverse proxy, handling traffic from the internet to internal Docker apps.

Here’s my problem: No matter what I set under the GOTIFY_URLS environment variable, it seems that iGotify replaces it with https://gotify, even though Gotify is listening on port 80 — in other words, over plain HTTP.

My question is: What do I need to change in the environment settings so that the connection to the Gotify service is made internally over port 80?

In short: I want iGotify to connect to http://gotify, not https://gotify.

**Comment by androidseb25** [comment]
Ok, that's the way to make a local instance.

You need to configure everything over the app and you will get all necessary informations for the environments.
Take a look at the screenshot and the example env below.

For you're Gotify URL you enter you're `gotify.domain.org` in the field

```
GOTIFY_URLS: 'http://192.168.178.199:8680'
GOTIFY_CLIENT_TOKENS: 'C1RXFpp_aHrmWyg'
SECNTFY_TOKENS: 'NTFY-DEVICE-hJGWiXsfAFC4nWJoNJksi9qKw23bcGAXAmNwqIaJlqh8V2mWdmHc27'
```

<img width="250" alt="Image" src="https://github.com/user-attachments/assets/5f7a9374-29d4-4661-a259-15b7f1bd8bab" />
<img width="250" alt="Image" src="https://github.com/user-attachments/assets/76955ab2-36fe-49ba-b254-fee12763c16d" />
<img width="250" alt="Image" src="https://github.com/user-attachments/assets/f59fb101-adc1-4f84-9cac-2856d71e721b" />

**Comment by Jan-Kruis** [comment]
Hi Sebastion,

I think there's a miscommunication about what exactly my problem is.

My problem isn't with the communication between the iGotify iPhone app and the GOTIFY Docker app, but between the iGOTIFY Docker app and the GOTIFY Docker services.

The connection between my iPhone and the GOTIFY app works fine.
The iPhone is able to retrieve messages from the GOTIFY Docker services, but I'm not receiving any notifications from GOTIFY via iGOTIFY on my phone.
I believe this is because the iGOTIFY services can't connect to the GOTIFY services, as shown in the log files I sent.

BTW I'm using the local instance on my iPhone and an HTTPS connection.

My problem is that the iGOTIFY Docker service can't communicate with the GOTIFY Docker service.
Both services are running in a one stack. See my docker-compos.yaml file.

I think the problem is that iGOTIFY converts the various entries I enter in the GOTIFY_URLS to a sort of default value like https://gotify.
Since the GOTIFY application only listens internally for http, iGOTIFY is unable to connect.

What could be causing iGOTIFY to constantly call the https server and the http version?



docker exec -it igotify sh -c "curl http://gotify:80/health"

{"health":"green","database":"green"}



docker exec -it igotify env | grep -E "GOTIFY|NTFY"

GOTIFY_URLS=http://gotify:80
GOTIFY_CLIENT_TOKENS=C8b............
SECNTFY_TOKENS=NTFY-DEVICE-u2h...............



docker logs igotify

Database is created: True
Gotify Server: 'https://gotify' is not available try to reconnect in 10s.
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://[::]:8080
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: /app
warn: Microsoft.AspNetCore.HttpsPolicy.HttpsRedirectionMiddleware[3]
      Failed to determine the https port for redirect.
Reconnecting...
Gotify Server: 'https://gotify' is not available try to reconnect in 10s.
Reconnecting...
Gotify Server: 'https://gotify' is not available try to reconnect in 10s.
Reconnecting...

**Comment by androidseb25** [comment]
can you take a look at you're users.db file in the data folder from iGotify in docker?

maybe there is a misconfigure and there stands https in it.

the users.db is a sqlite3 db

**Comment by Jan-Kruis** [comment]
Hi Sebastion,

thanks by removing the user.db 

this is now de igotufy log file

Database is created: True
Gotify - Url: http://gotify:80
Is Gotify - Url available: yes
SecNtfy Server - Url: https://api.secntfy.app
Is SecNtfy Server - Url available: yes
Client - Token: C8bRPRbvPtnRUK7
Client connecting...
Done!
info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://[::]:8080
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Production
info: Microsoft.Hosting.Lifetime[0]
      Content root path: /app
warn: Microsoft.AspNetCore.HttpsPolicy.HttpsRedirectionMiddleware[3]
      Failed to determine the https port for redirect.

so it is working now
